en/middleware/mavlink.md

```diff
@@ -2,200 +2,419 @@
+The `development/mavlink.h` header is also generated at build time (where "development" in the path comes from the XML definition file name).
+:::
+
+Then copy the streaming class definition below into the file:
+
+```cpp
+class MavlinkStreamBatteryStatusDemo : public MavlinkStream
 {
 public:
+    static MavlinkStream *new_instance(Mavlink *mavlink)
+    {
+        return new MavlinkStreamBatteryStatusDemo(mavlink);
+    }
     const char *get_name() const
     {
-        return MavlinkStreamCaTrajectory::get_name_static();
+        return MavlinkStreamBatteryStatusDemo::get_name_static();
     }
     static const char *get_name_static()
     {
-        return "CA_TRAJECTORY";
+        return "BATTERY_STATUS_DEMO";
     }
     static uint16_t get_id_static()
     {
-        return MAVLINK_MSG_ID_CA_TRAJECTORY;
+        return MAVLINK_MSG_ID_BATTERY_STATUS_DEMO;
     }
     uint16_t get_id()
     {
         return get_id_static();
     }
-    static MavlinkStream *new_instance(Mavlink *mavlink)
-    {
-        return new MavlinkStreamCaTrajectory(mavlink);
-    }
     unsigned get_size()
     {
-        return MAVLINK_MSG_ID_CA_TRAJECTORY_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
+        return MAVLINK_MSG_ID_BATTERY_STATUS_DEMO_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
     }
 
 private:
-    uORB::Subscription _sub{ORB_ID(ca_trajectory)};
+    //Subscription to array of uORB battery status instances
+    uORB::SubscriptionMultiArray<battery_status_s, battery_status_s::MAX_INSTANCES> _battery_status_subs{ORB_ID::battery_status};
+    // SubscriptionMultiArray subscription is needed because battery has multiple instances.
+    // uORB::Subscription is used to subscribe to a single-instance topic
 
     /* do not allow top copying this class */
-    MavlinkStreamCaTrajectory(MavlinkStreamCaTrajectory &);
-    MavlinkStreamCaTrajectory& operator = (const MavlinkStreamCaTrajectory &);
+    MavlinkStreamBatteryStatusDemo(MavlinkStreamBatteryStatusDemo &);
+    MavlinkStreamBatteryStatusDemo& operator = (const MavlinkStreamBatteryStatusDemo &);
 
 protected:
-    explicit MavlinkStreamCaTrajectory(Mavlink *mavlink) : MavlinkStream(mavlink)
+    explicit MavlinkStreamBatteryStatusDemo(Mavlink *mavlink) : MavlinkStream(mavlink)
     {}
 
-    bool send() override
-    {
-        struct ca_traj_struct_s _ca_trajectory;    //make sure ca_traj_struct_s is the definition of your uORB topic
+	bool send() override
+	{
+		bool updated = false;
 
-        if (_sub.update(&_ca_trajectory)) {
-            mavlink_ca_trajectory_t _msg_ca_trajectory;  //make sure mavlink_ca_trajectory_t is the definition of your custom MAVLink message
+		// Loop through _battery_status_subs (subscription to array of BatteryStatus instances)
+		for (auto &battery_sub : _battery_status_subs) {
+            // battery_status_s is a struct that can hold the battery object topic
+			battery_status_s battery_status;
```

Comment:
Original version imported this as a struct - ie. `struct battery_status_s battery_status;` or similar. Doesn't seem to be necessary to declare that type.